### ISSUE TITLE
--ext option

### ISSUE THREAD

**Issue by RyanZim** [postcss/postcss-cli#79]
Refs: https://github.com/postcss/postcss-cli/issues/78

We need an `--ext` option to allow setting the file extension when using `--dir`.

**Edit:** Just realized this will also solve the problem of people wanting to output to `.min.css`.

---

Should we rename `.pcss` to `.css` automatically? I think this is a bit too magical.

CC: @michael-ciniawsky @ai

**Comment by ai** [comment]
I found a case when you need to keep `.pcss` — postcss-sorting and other source-to-source transformation. So right now I am against auto-renaming.

**Comment by ai** [comment]
But maybe we could use `--ext pcss` for `.pcss` → `.pcss` transformation?

**Comment by RyanZim** [comment]
> I am against auto-renaming.

Me too; what's so bad about `--ext .css`?

**Comment by michael-ciniawsky** [comment]
@RyanZim I'm try to implement this with the few changes I'm still working on, but I had other work the last days. 80%

Personally also against autorenaming bacause of stylelint, sorting, recommenting with postcss-scss etc. The flag approach is fine imho

**Comment by michael-ciniawsky** [comment]
```postcss --ext|-e '.extname'``` or ```postcss src/**/*.pcss -o dest/**/*.css``` but the latter is confusing 🙃

**Comment by aesopturtle** [comment]
@michael-ciniawsky I agree. `--ext` seems cleaner to me too :)

**Comment by RyanZim** [comment]
That settles the autorename question.

---

> the latter is confusing

Agreed, let's stick to `--ext`.

---

One question: should we ensure that `--ext` starts with a period? That would allow for 

    postcss *.pcss --ext css -d out

I can't see any drawbacks to this.

**Comment by ai** [comment]
`--ext css` is OK for me

**Comment by michael-ciniawsky** [comment]
ok I implement `--ext` for now we can refcator and final review if needed :)

**Comment by RyanZim** [comment]
@michael-ciniawsky OK, If you prefer, I'd be more that glad to do it.

**Comment by michael-ciniawsky** [comment]
@RyanZim Alright if I don't push until tomorrow, bc of the social work line or whatever :) just go ahead, the problem is that I need to rearrange the code a bit to get 


```bash
postcss --config path/to/config [...options] blub  
```
**postcss.config.js**
```js
export default ({ options, file, env }) => ({
   parser:  file.extname === '.sss' ? 'sugarss' : false
   plugins: {
     'postcss-import': { root: file.dirname },
     'postcss-blub': options.blub,
     'cssnano': env === 'production' ? {} : false 
   }
})
```

Do *each* file must be exposed to `postcss-load-config`, mainly adding the config stuff to `processCSS` and assign them with the command line config. Also options can always be assigned as long as there is no  ```--use postcss-plugin``` specified.

**Comment by RyanZim** [comment]
Is running each file through `postcss-load-config` a WIP?

**Comment by michael-ciniawsky** [comment]
More or less yes if you have a working solution share it

**Comment by RyanZim** [comment]
I haven't done anything, just wondered if you were working on it.

**Comment by michael-ciniawsky** [comment]
Implemented => Can be closed :)

**Comment by RyanZim** [comment]
The current implementation doesn't ensure that `--ext` starts with a period; will implement.

**Comment by RyanZim** [comment]
Fixed in 43fc5db958e92ac4c746a2283dfc676b8b8e92aa
